linera-execution/src/system.rs

```diff
@@ -1061,6 +1061,23 @@ where
             .map_err(|_| SystemExecutionError::BlobNotFoundOnRead(blob_id))
             .map(Into::into)
     }
+
+    pub async fn assert_blob_exists(
+        &mut self,
+        blob_id: BlobId,
+    ) -> Result<(), SystemExecutionError> {
+        if self
+            .context()
+            .extra()
+            .contains_blob(blob_id)
+            .await
+            .map_err(|_| SystemExecutionError::BlobNotFoundOnRead(blob_id))?
```

Comment:
That means we returned an error, rather than `true` or `false`? Shouldn't we fail with _that_ error, then? What do we do in other places where there's some storage error?

Reply:
The error types are a bit tricky in the current state, but you're right, we should treat `false` and actual errors differently. Let me see what I can do